### DISCUSSION TITLE
Updating LazyVim causes permissions errors

### DISCUSSION THREAD

**Discussion by scottrohrig** [LazyVim/LazyVim#3545] [Q&A] [answered]
I regularly get `EACCESS: permission denied` errors for various plugins and lazyvim after updating.

I can fix it by going to the plugin dir (`~/.local/share/nvim/lazy/*`) and setting the permissions to my user, however it comes back every once in a while and is becoming a bit of a problem. I should also mention that this is not isolated to one machine. It happens on a few of my different remote machines and has happened locally (though I haven't had it locally for a while).

I'm sure I'm doing something wrong in my install (clone. However, I did install and setup from scratch and it happened again. Has anyone else experienced similar issues and can point me to better identify a root cause?

As far as dependencies:
- I am running nodejs@lts
- have fzf, rg & some others installed

I've run :LazyHealth and there doesn't seem to be any issues.

**Comment by rubiin** [comment, marked as answer]
something similar here https://github.com/LazyVim/LazyVim/discussions/3634

**Comment by mirsella** [comment]
Do you run nvim with sudo ?

you mentioned the dir permissions would change to another user, which user ? root i guess ?

**Reply by rubiin** [reply]
You can try all the solutions here https://stackoverflow.com/questions/2642836/git-error-unable-to-append-to-git-logs-refs-remotes-origin-master-permission

**Reply by rubiin** [reply]
sudo chown  - as last resort in your case

**Reply by mirsella** [reply]
@rubiin thanks but i've got no problem, i'm trying to help OP

**Reply by dpetka2001** [reply]
Maybe he's using nvim from docker containers and uses `~/.local/share/nvim/` as a common place to install configuration and mounts that to the docker container, so that all plugins are available to the containers, but container has different permissions from his regular user? Just a guess.

**Reply by mirsella** [reply]
Yes a lot of things can happen, OP should think what he has on all of his setup which is uncommon and can change the perms of the dir.
maybe try default lazyvim to see if it's not the lazyvim config, or if it's another thing on the system

**Reply by rubiin** [reply]
True. Permissions issues can be caused by many things . I think `chown` might be a silver bullet but this will cause some issues in future IMO

**Reply by scottrohrig** [reply]
Thanks mirsella, I don't run nvim as sudo unless I need to edit files for another or system files (eg, /etc/hosts). 
Thanks rubiin, I have solved this issue in the past with that method. It's strange that it has happened on multiple separate machines though. 
dpetka, not using docker, but thanks for the tip.

I cleared my lazy plugins located at `~/.local/share/nvim/lazy`. The permissions issue is no longer happening due to removing those directories. But that doesn't necessarily identify the root cause.
